joomla/joomla-cms
# small regression: [com_menus] items view: use empty for all menu instead of *

#### Summary of Changes

The "All Menu Items" was implemented in https://github.com/joomla/joomla-cms/pull/10190 (which used empty as the value for the "- Select Menu -" option), but after the menus ACL was implemented in https://github.com/joomla/joomla-cms/pull/9814 (which changed to `*` the value for the "- Select Menu -" option).

So, this PR goes back to (empty) as the value for the "- Select Menu -" option.

This allows to use `/administrator/index.php?option=com_menus&view=items&menutype=`URL (without a 403) as a fallback (in case the menu type is not found or something). This URL lists the all menu items list view.
#### Testing Instructions
1. Use latest staging
2. Go to `/administrator/index.php?option=com_menus&view=items&menutype=` you will get a 403 error
3. Now apply patch
4. Go to `/administrator/index.php?option=com_menus&view=items&menutype=` you'll be redirect to "All Menu Items" with no 403 error
5. Navigate through the menus and create/Edit menu items and check if you're properly redirected after without issues
6. Test batch and other operations also to confirm all is ok.

@infograf768 @bembelimen since you're the original creators of the two PR mentioned please test if all work as it should.
#### Observations

After this PR if you use:
- `/administrator/index.php?option=com_menus&view=items` you'll be redirect to your current state selected menu item ("All Menu Items" if none).
- `/administrator/index.php?option=com_menus&view=items&menutype=` you'll be redirect to "All Menu Items"
- `/administrator/index.php?option=com_menus&view=items&menutype=validmenutype` you'll be redirect to your valid menu type
- `/administrator/index.php?option=com_menus&view=items&menutype=invalidmenutype` you'll get a error the menu is not found
- `/administrator/index.php?option=com_menus&view=items&menutype=validmenutypebutnopermission` you'll get a error not autorized
